qiskit/transpiler/passes/layout/noise_adaptive_layout.py

```diff
@@ -71,6 +72,21 @@ def __init__(self, backend_prop):
         else:
             self.target = None
             self.backend_prop = backend_prop
+
+            if coupling_map:
+                # A backend might have more properties than qubits/gates in the configuration. This is a
+                # problem that the Target path should handle differently (by solving that possible
+                # inconsistency internally). For the non-target path, this is a possible solution.
+                # See https://github.com/Qiskit/qiskit/issues/7677
+                self.backend_prop.gates = filter(
+                    lambda ginfo: ginfo.gate == "cx"
+                    and tuple(ginfo.qubits) in coupling_map.graph.edge_list(),
+                    backend_prop.gates,
+                )
+                self.backend_prop.qubits = backend_prop.qubits[
```

Comment:
Does this modify the `BackendProperties` object for other passes that are using it, do we need a copy here before modifying? For the default preset pass managers, I don't think it really matters because the only things after this that access the backend properties are potentially `VF2PostLayout`, `Optimize1qGatesDecomposition`, and `UnitarySynthesis` (I might have missed one from memory) and those only look up error rates in the properties.

Reply:
Good point. I made the copy only under this condition, so I dont penalize the most general case. Done in 1a375dfbeeb9640a2d9782498e0fa5076f768ef2